README.md

```diff
@@ -23,107 +23,115 @@ api:
   insecure: true
 ```
 
-Start Traefik:
+Start Traefik v3:
 
-```bash
+```sh
 docker run -d -p 8080:8080 -p 80:80 \
--v $PWD/traefik.yml:/etc/traefik/traefik.yml \
--v /var/run/docker.sock:/var/run/docker.sock \
-traefik:v2.5
-```
+  -v $PWD/traefik.yml:/etc/traefik/traefik.yml \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  traefik:v3
+```  
 
-Start a backend server, named `test`:
+Start a backend server using the `traefik/whoami` image:
 
-```bash
+```sh
 docker run -d --name test traefik/whoami
 ```
 
-And finally, you can access to your `whoami` server throught Traefik, on the domain name `test.docker.localhost`:
+Access the whoami Service through Traefik via the defined rule `test.docker.localhost`:
 
 ```console
-# $ curl --header 'Host:test.docker.localhost' 'http://localhost:80/'
 $ curl test.docker.localhost
-Hostname: 390a880bdfab
+Hostname: 0693100b16de
 IP: 127.0.0.1
-IP: 172.17.0.3
+IP: ::1
+IP: 192.168.215.4
+RemoteAddr: 192.168.215.3:57618
 GET / HTTP/1.1
 Host: test.docker.localhost
-User-Agent: curl/7.65.3
+User-Agent: curl/8.7.1
 Accept: */*
 Accept-Encoding: gzip
-X-Forwarded-For: 172.17.0.1
+X-Forwarded-For: 192.168.215.1
 X-Forwarded-Host: test.docker.localhost
 X-Forwarded-Port: 80
 X-Forwarded-Proto: http
-X-Forwarded-Server: 7e073cb54211
-X-Real-Ip: 172.17.0.1
+X-Forwarded-Server: 8a37fd4f35fb
+X-Real-Ip: 192.168.215.1
 ```
 
-The web UI [http://localhost:8080](http://localhost:8080) will give you an overview of the routers, services, and middlewares.
+Access the Traefik Dashboard:
 
-![Web UI](https://raw.githubusercontent.com/traefik/traefik/v2.0/docs/content/assets/img/webui-dashboard.png)
+Open your web browser and navigate to `http://localhost:8080` to access the Traefik dashboard.
+This will provide an overview of routers, services, and middlewares.
 
-## Traefik v1 - Example usage
+![Dashboard UI](https://raw.githubusercontent.com/traefik/traefik/v3.2/docs/content/assets/img/webui-dashboard.png)
 
-Grab a [sample configuration file](https://raw.githubusercontent.com/traefik/traefik/v1.7/traefik.sample.toml) and rename it to `traefik.toml`. Enable `docker` provider and web UI:
+## Traefik v2 - Example usage
 
-```toml
-## traefik.toml
+Enable `docker` provider and dashboard UI:
 
-# API and dashboard configuration
-[api]
+```yml
+## traefik.yml
 
 # Docker configuration backend
-[docker]
-  domain = "docker.localhost"
+providers:
+  docker:
+    defaultRule: "Host(`{{ trimPrefix `/` .Name }}.docker.localhost`)"
```

Comment:
@sheddy-traefik @emilevauge One question, we are promoting the default rule configuration here to have the router created automatically/magically for the whoami backend (an option which IMHO is not widely used). I wonder if we should change this and add the router labels to the whoami container deployment? wdyt?

Reply:
i don't think it matters whether it is widely used or not, what matters is that is it recommended? is it possible? are there any drawbacks to using this over router labels. Also we have to keep in mind that this is a demo and not something people look at to model their production environments after. 

Let me know what you think

Reply:
This is working and there are no drawbacks (just lack of controls), you are right, for a demo I think this is ok.